backend/clap/proto/data_store.proto

```diff
@@ -7,113 +7,129 @@ option java_outer_classname = "DataStoreProto";
 
-    repeated Assessor assessors = 5;
-    repeated Rating   ratings   = 6;
+  repeated Assessor assessors = 5;
+  repeated Rating   ratings = 6;
 }
 
 message Assessor {
-    string first_name  = 1;
-    string last_name   = 2;
-    int32  assessor_id = 3;
+  string first_name = 1;
+  string last_name = 2;
+  int32  assessor_id = 3;
 }
 
 message Rating {
-    int32      rating_id   = 1;
-    bool       is_draft    = 2;
-    int32      assessor_id = 3;
-    RatingType type        = 4;
+  int32      rating_id = 1;
+  bool       is_draft = 2;
+  int32      assessor_id = 3;
+  RatingType type = 4;
 }
 
 message DetailsSubmissionResponse {
-    int32        team_size   = 1;
-    string       finish_date = 2;
-    ProjectState status      = 3;
-    string       budget      = 4;
-    string       description = 5;
-    AppReport    report      = 6;
+  int32        team_size = 1;
+  string       finish_date = 2;
+  ProjectState status = 3;
+  string       budget = 4;
+  string       description = 5;
+  AppReport    report = 6;
 }
 
 message AppReport {
-    bool is_draft                 = 1;
-    string report_submission_date = 2;
-    string project_goals          = 3;
-    string organisation_structure = 4;
-    string division_of_work       = 5;
-    string project_schedule       = 6;
-    string attatchments           = 7;
+  bool is_draft = 1;
+  string report_submission_date = 2;
+  string project_goals = 3;
+  string organisation_structure = 4;
+  string division_of_work = 5;
+  string project_schedule = 6;
+  string attatchments = 7;
 }
 
 message Criterion {
-    int32 criterion_id = 1;
-    string name        = 2;
-    string description = 3;
-    string area        = 4;
-    string criteria    = 5;
-    string subcriteria = 6;
+  int32 criterion_id = 1;
+  string name = 2;
+  string description = 3;
+  string area = 4;
+  string criteria = 5;
+  string subcriteria = 6;
 }
 
 message PartialRating {
-    int32  partial_rating_id = 1;
-    int32  criterion_id      = 2;
-    int32  points            = 3;
-    string justification     = 4;
+  int32  partial_rating_id = 1;
+  int32  criterion_id = 2;
+  int32  points = 3;
+  string justification = 4;
 
-    string modified          = 5;
-    int32  modified_by       = 6;
+  string modified = 5;
+  int32  modified_by = 6;
 }
 
 message AssessorRatings {
-    int32 assessor_id = 1;
-    string first_name = 2;
-    string last_name  = 3;
-    repeated PartialRating partial_ratings = 4;
+  int32 assessor_id = 1;
+  string first_name = 2;
+  string last_name = 3;
+  repeated PartialRating partial_ratings = 4;
 }
 
 message RatingsSubmissionResponse {
-    repeated Criterion criteria = 1;
+  repeated Criterion criteria = 1;
 
-    repeated AssessorRatings individual = 2;
-    AssessorRatings initial    = 3;
-    AssessorRatings final      = 4;
+  repeated AssessorRatings individual = 2;
+  AssessorRatings initial = 3;
+  AssessorRatings final = 4;
+}
+
+message RatingRequest {
```

Comment:
As we agreed today, the proto file should reflect the following HTTP endpoints:
- POST /submissions/{submissionId}/ratings
with body: {
    ratingType: RatingType;  
}
which will create an empty rating for the given submission. If it's a individual rating, it should retrieve the user ID from the token (as we'll change how we handle authorization). If it's an initial/final rating, it should assign the rating to the leading assessor of the submission.

- POST /partialRatings
with body: {
    criterionId: int?;
    ratingId: int?;
    partialRatingId: int?;
    justification: string;
    points: int;
}
which will perform an upsert of the partial rating. Exactly one of the parameters `ratingId` and `partialRatingId` must be specified. 
If it's `ratingId`, a new partial rating should be created under the specified rating and criterion. 
If it's `partialRatingId`, the partial rating should be updated with the justification and points.

- POST /ratings/{ratingId}/submit which actually doesn't require any body (if you're concerned with POST with empty body, I'm fine to change the interface) and will submit a rating from the draft to the non-draft state. It also includes authorization:
    - only the owner of the individual rating is authorized to submit it
    - only the leading assessor of the initial/final rating is authorized to submit it
and performs a check if all the required partial ratings are present for the rating.

CC: @Veermove pls check if there are no gaps in this contract

PS: the interfaces are something between Java and TS, I hope you will find out what I meant ('?' stands for optional property)